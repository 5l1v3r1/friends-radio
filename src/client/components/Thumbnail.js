import React from "react"
import glamorous from "glamorous"

export const Wrapper = glamorous.div({
  background: "rgba(0, 0, 0, .5)",
  opacity: 0,
  transition: "all ease-in-out 300ms",
  ":hover": {
    opacity: 1,
  },
  width: "100%",
  height: "100%",
})

const Thumbnail = ({src, children, active, ...otherProps}) => {
  const thumbnailStyle = {
    backgroundImage: `url(${src})`,
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    backgroundPosition: "center center",
  }

  let activeStyle = undefined
  if (active === true) activeStyle = {opacity: 1}
  else if (active === false) activeStyle = {opacity: 0}

  if (otherProps.style) Object.assign(thumbnailStyle, otherProps.style)

  return (
    <div {...otherProps} style={thumbnailStyle}>
      <Wrapper style={activeStyle}>
        {children}
      </Wrapper>
    </div>
  )
}

export default Thumbnail
