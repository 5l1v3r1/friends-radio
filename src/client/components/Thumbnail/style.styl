.wrapper
  background rgba(0, 0, 0, .5)
  opacity 0
  transition all ease-in-out 300ms
  &:hover
    opacity 1
  width 100%
  height 100%
