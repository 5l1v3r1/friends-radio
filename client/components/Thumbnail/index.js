import React from 'react'

import style from './style.styl'

const Thumbnail = ({src, children, ...otherProps}) => {
  const thumbnailStyle = {
    backgroundImage: `url(${src})`,
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center center'
  }

  if (otherProps.style)
    Object.assign(thumbnailStyle, otherProps.style)

  return (
    <div {...otherProps} style={thumbnailStyle}>
      <div className={style.wrapper}>
        {children}
      </div>
    </div>
  )
}

export default Thumbnail
