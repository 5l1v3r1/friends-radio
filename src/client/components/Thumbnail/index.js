import React from 'react'

import style from './style.styl'

const Thumbnail = ({src, children, active, ...otherProps}) => {
  const thumbnailStyle = {
    backgroundImage: `url(${src})`,
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center center'
  }

  let activeStyle = undefined
  if (active === true)
    activeStyle = {opacity: 1}
  else if (active === false)
    activeStyle = {opacity: 0}

  if (otherProps.style)
    Object.assign(thumbnailStyle, otherProps.style)

  return (
    <div {...otherProps} style={thumbnailStyle}>
      <div className={style.wrapper} style={activeStyle}>
        {children}
      </div>
    </div>
  )
}

export default Thumbnail
