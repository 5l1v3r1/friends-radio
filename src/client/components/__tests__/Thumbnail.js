import React from "react"
import {shallow} from "enzyme"

import Thumbnail, {Wrapper} from "../Thumbnail"

test("it should pass props correctly", () => {
  const wrapper = shallow(
    <Thumbnail
      src="foobar"
      customProps
      custom="withArgs"
      style={{foo: "bar"}}
    />,
  )
  const props = wrapper.find("div").props()
  expect(props).toMatchSnapshot()
})

test("active/inactive", () => {
  const active = shallow(<Thumbnail active={true} />)
  const inactive = shallow(<Thumbnail active={false} />)
  expect(active.find(Wrapper).props().style).toEqual({opacity: 1})
  expect(inactive.find(Wrapper).props().style).toEqual({opacity: 0})
})

it("should pass children", () => {
  const wrapper = shallow(
    <Thumbnail>
      <p>Hello</p>
      <span>World</span>
    </Thumbnail>,
  )
  expect(wrapper.find(Wrapper).children()).toMatchSnapshot()
})
