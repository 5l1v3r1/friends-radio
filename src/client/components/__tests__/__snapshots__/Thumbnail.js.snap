// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should pass props correctly 1`] = `
Object {
  "children": <glamorous(div)
    style={undefined}
/>,
  "custom": "withArgs",
  "customProps": true,
  "style": Object {
    "backgroundImage": "url(foobar)",
    "backgroundPosition": "center center",
    "backgroundRepeat": "no-repeat",
    "backgroundSize": "cover",
    "foo": "bar",
  },
}
`;

exports[`should pass children 1`] = `
Array [
  <p>
    Hello
</p>,
  <span>
    World
</span>,
]
`;
